src/components/content/exercises/exercise-generation-wizard/difficulty.tsx

```diff
@@ -0,0 +1,90 @@
+import { ExerciseGenerationDifficulty } from './generate-prompt'
+import { WizardPageProps } from './wizard-page-props'
+import { MenuButton, MenuItem } from '../menu-button'
+import { useLoggedInData } from '@/contexts/logged-in-data-context'
+import { LoggedInData } from '@/data-types'
+import { replacePlaceholders } from '@/helper/replace-placeholders'
+
+interface DifficultyProps extends WizardPageProps {
+  difficulty: ExerciseGenerationDifficulty | null
+  setDifficulty: (level: ExerciseGenerationDifficulty | null) => void
+  learningGoal: string
+  setLearningGoal: (goal: string) => void
+}
+
+export const Difficulty: React.FC<DifficultyProps> = ({
+  difficulty,
+  setDifficulty,
+  learningGoal,
+  setLearningGoal,
+  onNext,
+  isSummary,
+}) => {
+  const { strings } = useLoggedInData() as LoggedInData
```

Comment:
if you are sure that at this point the user will always be logged in you could also use `!` to shorten this:
```ts
  const { strings } = useLoggedInData()!
```

If you are not entirely sure that's what I usually use:
```ts
  const loggedInData = useLoggedInData()
  if(!loggedInData) return null // or handle somehow
```

and to simplify your component I'd use:
```suggestion
  const difficultyStrings = useLoggedInData()!.strings.ai.exerciseGeneration.difficulty
  …
```
(here and in the other comps)